File: src/main/java/factory/CourseScheduleViewFactory.java
package factory;

import app.FXMLPaths;
import control.schedule.CourseScheduleController;
import control.schedule.QuarterScheduleController;
import javafx.scene.Node;

import java.io.IOException;

public class CourseScheduleViewFactory extends ViewFactory<CourseScheduleController> {

    private final CourseScheduleController controller;

    public CourseScheduleViewFactory(CourseScheduleController controller) {
        super(FXMLPaths.CourseSchedule());
        this.controller = controller;
    }

    @Override
    public Node load() throws IOException {
        Node q1 = new DualWeekScheduleViewFactory<>(controller.getFirstQuarterController()).load();
        Node q2 = new DualWeekScheduleViewFactory<>(controller.getSecondQuarterController()).load();

        Node n = super.load(controller);

        controller.setFirstQuarterContent(q1);
        controller.setSecondQuarterContent(q2);

        return n;
    }

    @Override
    public Node load(CourseScheduleController controller) throws IOException {
        throw new UnsupportedOperationException();
    }
}

File: src/main/scala/factory/CourseScheduleViewFactory.scala
package factory

import java.io.IOException

import app.FXMLPaths
import control.schedule.{CourseScheduleController, QuarterScheduleController, WeekScheduleController}
import javafx.scene.Node

class CourseScheduleViewFactory(controller: CourseScheduleController) extends
    ViewFactory[CourseScheduleController](FXMLPaths.CourseSchedule) {

    @throws[IOException]
    override def load: Node = {
        //TODO check type inference
        val q1 = new DualWeekScheduleViewFactory[QuarterScheduleController, WeekScheduleController, WeekScheduleController](controller.getFirstQuarterController).load
        val q2 = new DualWeekScheduleViewFactory[QuarterScheduleController, WeekScheduleController, WeekScheduleController](controller.getSecondQuarterController).load

        val n = super.load(controller)

        controller.setFirstQuarterContent(q1)
        controller.setSecondQuarterContent(q2)
        n
    }

    @throws[UnsupportedOperationException]
    override def load(controller: CourseScheduleController) = throw new UnsupportedOperationException
}
